pkg/ingester/ingester.go

```diff
@@ -314,7 +326,8 @@ func New(cfg Config, limits *validation.Overrides, activeGroupsCleanupService *u
 		return nil, err
 	}
 	i.ingestionRate = util_math.NewEWMARate(0.2, instanceIngestionRateTickInterval)
-	i.metrics = newIngesterMetrics(registerer, cfg.ActiveSeriesMetrics.Enabled, i.getInstanceLimits, i.ingestionRate, &i.inflightPushRequests)
+	i.metrics = newIngesterMetrics(registerer, cfg.ActiveSeriesMetrics.Enabled, i.getInstanceLimits, i.ingestionRate, &i.inflightPushRequests,
```

Comment:
Looks a bit cumbersome the way you're doing the metrics wiring. I think you should simply expose the metrics directly in the `UtilizationBasedLimiter`. You can pass the registry to `NewUtilizationBasedLimiter()`. Since the limiter is a generic one, the metric name may have no `cortex_ingester_` prefix, then you can wrap the registered passed to `NewUtilizationBasedLimiter()` to add the `cortex_ingester_` prefix.

Reply:
I agree, this was a quick implementation for testing :)

Reply:
I will rewrite the PR based on the above feedback.